### PR TITLE
Save marks

### PR DESCRIPTION
**Pull request by INRokh**
Functionality save annotation results on the BE, known issue, if you put marks tooooo quickly app will crash.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -77,7 +78,9 @@ class App extends Component {
           </Switch>
         </BrowserRouter>
       );
-    } else {
+    }
+    // Logged in as user.
```
Comment (INRokh) [inline src/App.js:82]: added comments, and fix bug

File: src/components/Canvas.js [hunk 2]
```diff
@@ -46,11 +45,12 @@ class Canvas extends Component {
       ctx.drawImage(this.state.image, 0, 0, cv.width, cv.height);
       ctx.fillStyle = "#009900";
       ctx.font = "16px Courier";
-      for(let tag in this.props.marks){
-        for(let i = 0; i < this.props.marks[tag].length; i++){
-          const {x, y} = this.props.marks[tag][i];
+      for(let i in this.props.marks){
```
Comment (INRokh) [inline src/components/Canvas.js:48]: change structure from object(map) to array

File: src/components/ImageAnnotation.js [hunk 3]
```diff
@@ -1,31 +1,59 @@
...
         console.log("No tag selected.");
         return;
       };
-      let newMarksState = Object.assign({}, this.state.marks);
-      if(!newMarksState[this.state.selectedTag.title]){
-        newMarksState[this.state.selectedTag.title] = [];
-      };
-      newMarksState[this.state.selectedTag.title].push(coordinates);
-      this.setState({marks: newMarksState});
+      let newMarks = [...marks];
```
Comment (INRokh) [inline src/components/ImageAnnotation.js:19]: converts state data from API get response to API put request

File: src/components/ImageAnnotation.js [hunk 4]
```diff
@@ -1,31 +1,59 @@
...
+        newMarks[i].tag_id = tmp_id;
+        delete newMarks[i]._id;
+        for (let j in newMarks[i].coordinates) {
+          delete newMarks[i].coordinates[j]._id;
+        }
+      }
+      let selectedMark = null;
+      // Find selected tag in the marks and put its reference to
+      // selectedMark.
+      for (let i in newMarks) {
```
Comment (INRokh) [inline src/components/ImageAnnotation.js:32]: update marks state, when user add new info

**Comment by EllieChen-Git** [comment]
Thanks Nina, this is AWESOME :)

**Pull request merged**